src/content/1.7/development/architecture/migration-guide/strategy.md

```diff
@@ -0,0 +1,172 @@
+
+The Migration project started by migrating Configuration Form pages.
+
+[You can read more about it by clicking on this link][form-pages].
+
+The chosen structure to handle such pages was the following:
+
+1. A View layer: Twig template, rendering manually the Symfony form
+2. A controller layer: Symfony Controller empty of business logic (following the ["Thin Controller, Fat Model"][thin-ctrlers] principle), in charge of controlling the View, formatting data, handling security and routing
+3. In Model layer: Symfony Form is built and managed by a FormHandler
+4. Also part of the Model layer, DataProvider is in charge of data retrieval and saving the data
+
+Since the DataProvider interacts with ObjectModel, it belongs to [Adapter namespace][adapter-namespace].
+
+# Phase 2: listings
+
+For migrating Listing pages, the Grid component was built.
+
+[You can read more about it by clicking on this link][grid-pages].
+
+The chosen layer structure to handle such pages was the following:
+
+1. View layer: Twig template, using Grid Twig template
+2. Symfony Controller layer empty of business logic, in charge of controlling the View, formatting data, handling security and routing
+3. In Application layer, we have the Grid component which requires a GridDefinition configures the Grid and a GridFactory to build it
+4. Data is provided by a QueryBuilder, able to handle filters for search requests. This is the Model layer.
+
+If the Grid provides some actions, they are handled by dedicated Controller Actions.
+
+### How Grid interacts with the legacy Model
+
+The QueryBuilder ignores ObjectModel and use a Doctrine Connection to query directly the database. So there is no interaction with legacy code.
+
+However the Grid Actions use ObjectModel, for example to enable/disable a Category status. In the beginning we used Adapter classes to handle these, but we could see that once again we were coupling legacy code with migrated code.
+
+This is when we decided to introduce [CQRS][cqrs].
+
+# Phase 3: [CQRS][cqrs]
+
+We wanted to build a strong frontier between migrated code and legacy code. Since re-writing all the code would be too long, we needed to continue relying on the business logic legacy code to power PrestaShop.
+
+We however wanted to interact with it in a way that would allow us, in the future, to easily unplug it and replace it with a new implementation.
+
+So we decided to introduce [CQRS][cqrs] and [a Domain][domain].
+
+CQRS was introduced into PrestaShop architecture in the following manner:
+
+The Application layer (Forms, Controllers, Views ...) dispatches [Queries and Commands][cqrs] to the Domain to be handled, unaware of the inner workings of this domain.
+
+Inside the domain are QueryHandlers and CommandHandlers that handle dispatched Queries and Commands. For now, they do use PrestaShop legacy business logic to handle it. This means classes like ObjectModel, Cart, Product are still being used inside migrated pages.
```

Comment:
```suggestion
Inside the domain are QueryHandlers and CommandHandlers that handle dispatched Queries and Commands. For now, they use PrestaShop legacy business logic to handle it. This means classes like ObjectModel, Cart, Product are still being used inside migrated pages.
```

Reply:
I think this one has been made on purpose to accentuate that it's using PrestaShop legacy business logic.

Reply:
Indeed 😉

Reply:
:+1: